### PR TITLE
[16.0][ADD] - add stock_override_procurement

### PR DESCRIPTION
**Pull request by sbejaoui**
In odoo core modules there is often a method to provide the procurement values but not the attributes like location, product_id, etc. This technical module makes it possible to override them if they are passed in the values.

### CODE REVIEW

**Comment by sbejaoui** [comment]
cc/ @lmignon

File: stock_override_procurement/models/procurement_group.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        ]
+
+    @api.model
+    def _get_override_procurement(self, procurement):
+        values = procurement.values
+
+        def _get_override_value(field_name):
+            return (
+                values.get(field_name)
+                if values.get(field_name)
```
Comment (lmignon) [inline stock_override_procurement/models/procurement_group.py]:
```suggestion
                if field_name in values
```
required to allows to unset a value by override

**Review by lmignon — CHANGES_REQUESTED** [review]

**Review by lmignon — CHANGES_REQUESTED** [review]
Thank you @sbejaoui A useful addon. A little change required for a specific case...

**Review by lmignon — APPROVED** [review]
LGTM (Code review)
Thank you @sbejaoui

**Review by rousseldenis — APPROVED** [review]
Code review. LGTM

Hackish but will work great!

**Review by rvalyi — APPROVED** [review]
LGTM

**Pull request closed without merging**